### PR TITLE
Add support for diagram markers

### LINKED ISSUE #30: Problem marker support
We should implement a problem marker support for GLSP diagrams. (Similar to the problem marker support in Papyrus but for now  a  baseline and simplified solution suffices)

### PR DESCRIPTION
**Pull request by tortmayr**
From #216 :
Closes #30

This is an initial contribution for offering model validation support that shows validation info/warning/error messages as diagram decorators. I opened issues #213 and #214 for further improving this initial model validation support. I am happy to work on these issues if you think the suggested improvements make sense.

### CODE REVIEW

File: client/packages/glsp-sprotty/src/features/validation/validate.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+/**
+ * Command for handling SetMarkersActions
+ */
+@injectable()
+export class SetMarkersCommand extends Command {
+
+    static readonly KIND = 'setMarkers';
+
+    markers: Marker[];
```
Comment (tortmayr) [inline client/packages/glsp-sprotty/src/features/validation/validate.ts]: Why do we need this field?

File: client/packages/glsp-sprotty/src/features/validation/validate.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            issueMarker = new SIssueMarker();
+            issueMarker.type = "marker";
+            issueMarker.issues = new Array<SIssue>();
+            modelElement.add(issueMarker);
+        }
+
+        return issueMarker;
+    }
+
+    private createSIssue(marker: Marker): SIssue {
```
Comment (tortmayr) [inline client/packages/glsp-sprotty/src/features/validation/validate.ts]: Doesn't access any member variables so can be defined as function

File: client/packages/glsp-sprotty/src/features/validation/validate.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                break;
+            }
+        }
+
+        return issue;
+    }
+
+    undo(context: CommandExecutionContext): CommandResult {
+        this.markers = this.action.markers;
+        for (let marker of this.markers) {
```
Comment (tortmayr) [inline client/packages/glsp-sprotty/src/features/validation/validate.ts]: Should be `for (const marker`

File: client/packages/glsp-sprotty/src/features/validation/validate.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+            const modelElement: SModelElement | undefined = context.root.index.getById(marker.elementId);
+            if (modelElement instanceof SParentElement) {
+                for (let child of modelElement.children) {
```
Comment (tortmayr) [inline client/packages/glsp-sprotty/src/features/validation/validate.ts]: Should be `for (const child`

File: client/packages/glsp-sprotty/src/features/validation/validate.ts [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    redo(context: CommandExecutionContext): CommandResult {
+        return this.execute(context);
+    }
+
+    getMarkers(): Marker[] {
+        return this.markers;
+    }
+
```
Comment (tortmayr) [inline client/packages/glsp-sprotty/src/features/validation/validate.ts]: Please remove blank line

File: client/packages/glsp-sprotty/src/features/validation/validate.ts [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+            }
+        }
+        return context.root;
+    }
+
+    redo(context: CommandExecutionContext): CommandResult {
+        return this.execute(context);
+    }
+
+    getMarkers(): Marker[] {
```
Comment (tortmayr) [inline client/packages/glsp-sprotty/src/features/validation/validate.ts]: Why do we need that getter?  this.markers is public anyways

File: server/glsp-server/src/main/java/com/eclipsesource/glsp/server/actionhandler/RequestMarkersHandler.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+			elementsIDs = new String[] { modelState.getCurrentModel().getId() };
+		}
+		List<Marker> markers = new ArrayList<Marker>();
+		SModelIndex currentModelIndex = modelState.getCurrentModelIndex();
+		for (String elementID : elementsIDs) {
+			SModelElement modelElement = currentModelIndex.get(elementID);
+			markers.addAll(validator.validate(modelState, modelElement));
+		}
+
+		// TODO get element ID of root element on client side and send it (see
```
Comment (tortmayr) [inline server/glsp-server/src/main/java/com/eclipsesource/glsp/server/actionhandler/RequestMarkersHandler.java]: Seems like the TODOs are outdated.

File: server/glsp-api/src/main/java/com/eclipsesource/glsp/api/markers/MarkerKind.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package com.eclipsesource.glsp.api.markers;
+
+/**
+ * Defines kinds of model markers
+ *
+ */
+public class MarkerKind {
```
Comment (tortmayr) [inline server/glsp-api/src/main/java/com/eclipsesource/glsp/api/markers/MarkerKind.java]: Should be final

**Comment by tanjaem** [comment]
Thanks for your feedback! Should all be incorporated now.

**Pull request closed without merging**